src/btreemap.rs

```diff
@@ -1161,156 +1161,317 @@ mod test {
             let right_child = btree.load_node(root.child(1));
             assert!(right_child.is_full());
             let median_index = right_child.entries_len() / 2;
-            assert_eq!(right_child.key(median_index), &b(&[12]));
+            let median_key = key(12);
+            assert_eq!(right_child.key(median_index), &median_key);
 
             // Overwrite the value of the median key.
-            assert_eq!(btree.insert(b(&[12]), b(&[1, 2, 3])), Some(b(&[])));
-            assert_eq!(btree.get(&b(&[12])), Some(b(&[1, 2, 3])));
+            assert_eq!(btree.insert(median_key.clone(), value(123)), Some(value(0)));
+            assert_eq!(btree.get(&median_key), Some(value(123)));
 
             // The child has not been split and is still full.
             let right_child = btree.load_node(root.child(1));
             assert_eq!(right_child.node_type(), NodeType::Leaf);
             assert!(right_child.is_full());
         });
     }
+    btree_test!(
+        test_insert_overwrite_median_key_in_full_child_node,
+        insert_overwrite_median_key_in_full_child_node
+    );
 
-    #[test]
-    fn insert_overwrite_key_in_full_root_node() {
-        btree_test(|mut btree| {
+    fn insert_overwrite_key_in_full_root_node<K: TestKey, V: TestValue>() {
+        let (key, value) = (|i| K::build(i), |i| V::build(i));
+        run_btree_test(|mut btree| {
             for i in 1..=11 {
-                assert_eq!(btree.insert(b(&[i]), b(&[])), None);
+                assert_eq!(btree.insert(key(i), value(0)), None);
             }
 
             // We now have a root that is full and looks like this:
             //
-            // [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
+            // [1, 2, 3, 4, 5, (6), 7, 8, 9, 10, 11]
             let root = btree.load_node(btree.root_addr);
             assert!(root.is_full());
 
             // Overwrite an element in the root. It should NOT cause the node to be split.
-            assert_eq!(btree.insert(b(&[6]), b(&[4, 5, 6])), Some(b(&[])));
+            assert_eq!(btree.insert(key(6), value(456)), Some(value(0)));
 
             let root = btree.load_node(btree.root_addr);
             assert_eq!(root.node_type(), NodeType::Leaf);
-            assert_eq!(btree.get(&b(&[6])), Some(b(&[4, 5, 6])));
+            assert_eq!(btree.get(&key(6)), Some(value(456)));
             assert_eq!(root.entries_len(), 11);
         });
     }
+    btree_test!(
+        test_insert_overwrite_key_in_full_root_node,
+        insert_overwrite_key_in_full_root_node
+    );
 
-    #[test]
-    fn allocations() {
-        btree_test(|mut btree| {
+    fn allocations_without_split<K: TestKey, V: TestValue>() {
+        let key = |i| K::build(i);
+        run_btree_test(|mut btree| {
+            assert_eq!(btree.allocator.num_allocated_chunks(), 0);
+
+            assert_eq!(btree.insert(key(1), V::empty()), None);
+            assert_eq!(btree.allocator.num_allocated_chunks(), 1);
```

Comment:
FYI: here the size of value object may affect `num_allocated_chunks` therefore it has to be minimal (empty) size.